components/governance/org.wso2.carbon.governance.registry.extensions/src/org/wso2/carbon/governance/registry/extensions/executors/APIPublishExecutor.java

```diff
@@ -0,0 +1,301 @@
+	 * @return             Returns whether the execution was successful or not.
+	 */
+	@Override
+	public boolean execute(RequestContext context, String currentState, String targetState) {
+		Resource resource = context.getResource();
+
+		try {
+			String artifactString = RegistryUtils.decodeBytes((byte[]) resource.getContent());
+			String user = CarbonContext.getThreadLocalCarbonContext().getUsername();
+			OMElement xmlContent = AXIOMUtil.stringToOM(artifactString);
+			GenericArtifactManager manager = new GenericArtifactManager(
+					RegistryCoreServiceComponent.getRegistryService().getGovernanceUserRegistry(user, CarbonContext
+							.getThreadLocalCarbonContext().getTenantId()), ExecutorConstants.REST_SERVICE_KEY);
+
+			GenericArtifact api = manager.getGenericArtifact(context.getResource().getUUID());
+			publishData(api, xmlContent);
+
+		} catch (RegistryException e) {
+			log.error("Failed to publish service to API store ", e);
+			return false;
+		} catch (XMLStreamException e) {
+			log.error("Failed to convert service to xml content", e);
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Publish the data to API Manager
+	 *
+	 * @param api        API registry artifact.
+	 * @param xmlContent url Pattern value iterator.
+	 */
+	private void publishData(GenericArtifact api, OMElement xmlContent) throws RegistryException {
+
+		if (apimEndpoint == null || apimUsername == null || apimPassword == null) {
+			throw new RuntimeException(ExecutorConstants.APIM_LOGIN_UNDEFINED);
+		}
+
+		CookieStore cookieStore = new BasicCookieStore();
+		HttpContext httpContext = new BasicHttpContext();
+		httpContext.setAttribute(ClientContext.COOKIE_STORE, cookieStore);
+
+		Utils.authenticateAPIM(httpContext, apimEndpoint, apimUsername, apimPassword);
+		String publishEndpoint = apimEndpoint + ExecutorConstants.APIM_PUBLISH_URL;
+
+		// create a post request to addAPI.
+		HttpClient httpclient = new DefaultHttpClient();
+		HttpPost httppost = new HttpPost(publishEndpoint);
+
+		// Request parameters and other properties.
+		List<NameValuePair> params = getRequestParameters(api, xmlContent);
+
+		if (api.getAttribute(ExecutorConstants.SERVICE_ENDPOINT_URL) != null &&
+		    api.getAttribute(ExecutorConstants.SERVICE_ENDPOINT_URL).isEmpty()) {
+			log.warn(ExecutorConstants.EMPTY_ENDPOINT);
+		}
+
+		try {
+			httppost.setEntity(new UrlEncodedFormEntity(params, ExecutorConstants.DEFAULT_CHAR_ENCODING));
+
+			HttpResponse response = httpclient.execute(httppost, httpContext);
+
+			if (response.getStatusLine().getStatusCode() != 200) {
+				throw new RuntimeException(
+						"Failed : HTTP error code : " + response.getStatusLine().getStatusCode());
+			}
+
+		} catch (ClientProtocolException e) {
+			throw new RegistryException(ExecutorConstants.APIM_POST_REQ_FAIL, e);
+		} catch (UnsupportedEncodingException e) {
+			throw new RegistryException(ExecutorConstants.ENCODING_FAIL, e);
+		} catch (IOException e) {
```

Comment:
@Rajith90 Fixed with #261 (f6c902ac266cb33aa4c70b2de294b8d589d0eb8d). In this fix I have removed the catch clause for `UnsupportedClientException` as it is a sub class of `IOException` and the error message is same.